SDL3 upgrade

Berstanio opened the pull request:
This PR removes the jampad dependency and uses the lwjgl3 SDL bindings instead.
Some classes from jampad have been moved over. I didn't saw the need for jampad being a seperate project anymore, as lwjgl3 provides the bindings now.
This PR depends on the `3.4.0-SNAPSHOT` of lwjgl3.


By accident, I applied similar fixes as done in this pr: https://github.com/libgdx/gdx-controllers/pull/38
This pr is based of this pr: https://github.com/libgdx/gdx-controllers/pull/39

Berstanio committed:
feat(desktop): Initial translation

Berstanio committed:
feat(desktop): Minor fixes

Berstanio committed:
chore: License notice

MrStahlfelge commented:
Is there a problem for Lwjgl2 projects having both Lwjgl3 and 2 in the classpath?

PokeMMO commented:
There's a problem for lwjgl2 projects having lwjgl2 on the classpath.

MrStahlfelge commented:
Okay, I see the problem that people might not want to use Lwjgl3, but they can stick on older versions of libgdx and this library so this should not stop us.

However, I caused some conflicts by merging your other PR which I eventually tested.

MrStahlfelge commented:
We should raise the version to 2.3.x and add a new entry into the compatiblity matrix pointing out incompatibility with lwjgl2 projects.

Berstanio committed:
feat: Remove armeabi

Berstanio committed:
feat: libGDX 1.13.1 requires androidX

Berstanio committed:
fix: Bump random versions to fix weird android compile error

Berstanio committed:
fix: Fix removed method use `isCatchBackKey`

Berstanio committed:
feat: Run CI on JDK 17, required by android

Berstanio committed:
chore: Unused import

Berstanio committed:
fix: Fix old ordinal references

MrStahlfelge reviewed (commented):
Probably 'test/desktop' can be removed - it should not work any more?

'test/ios' does not compile, it seems like there is a dependency problem. Probably RoboVM needs to get updated to the version 1.13.x uses

Berstanio commented on build.gradle:
```diff
@@ -18,9 +18,9 @@ allprojects {
+        gdxVersion = '1.13.1'
         roboVMVersion = '2.3.19'
-        jamepadVersion = '2.26.5.0'
+        lwjgl3Version = '3.4.0-SNAPSHOT'
```
For the moment yes, so I would only merge this after a stable lwjgl3 release exists.

Yontipon replied:
I believe the stable LWJGL version we're waiting on is available now, if you want to move forward with this.

Berstanio replied:
Yes, I was waiting a bit for the usual Anti-Virtus fuss to settle, but 3.4.1 seems good now.

Berstanio committed:
feat: Remove lwjgl2 tests

Berstanio committed:
feat: Define lwjgl3 dependencies

Berstanio committed:
chore: Update robovm

Berstanio commented:
> Probably 'test/desktop' can be removed - it should not work any more?

Yes, removed it.

> 'test/ios' does not compile, it seems like there is a dependency problem.

A class was apparantly moved, fixed the import. I also upgraded roboVM in one go

Berstanio committed:
feat: Bump to lwjgl 3.4.1
